segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/chunklayer/GarbageCollector.java

```diff
@@ -447,21 +450,29 @@ public CompletableFuture<Void> processBatch(List<TaskInfo> batch) {
                 log.debug("{}: deleteGarbage - transaction is still active - re-queuing {}.", traceObjectId, infoToDelete.transactionId);
                 taskQueue.addTask(taskQueueName, infoToDelete);
             } else {
-                val f = executeSerialized(() -> processTask(infoToDelete), infoToDelete.name);
                 val now = currentTimeSupplier.get();
                 if (infoToDelete.scheduledTime > currentTimeSupplier.get()) {
                     futures.add(delaySupplier.apply(Duration.ofMillis(infoToDelete.scheduledTime - now))
-                            .thenComposeAsync(v -> f, storageExecutor));
+                            .thenComposeAsync(v -> executeSerialized(() -> processTask(infoToDelete), infoToDelete.name), storageExecutor));
```

Comment:
This line is not related to GC delay but I found this small fix from my other PR (https://github.com/pravega/pravega/pull/6597)